src/Components/WebAssembly/Authentication.Msal/src/Interop/AuthenticationService.ts

```diff
@@ -0,0 +1,363 @@
+import * as Msal from 'msal';
+import { StringDict } from 'msal/lib-commonjs/MsalTypes';
+import { ClientAuthErrorMessage } from 'msal/lib-commonjs/error/ClientAuthError';
+
+interface AccessTokenRequestOptions {
+    scopes: string[];
+    returnUrl: string;
+}
+
+interface AccessTokenResult {
+    status: AccessTokenResultStatus;
+    token?: AccessToken;
+}
+
+interface AccessToken {
+    value: string;
+    expires: Date;
+    grantedScopes: string[];
+}
+
+enum AccessTokenResultStatus {
+    Success = "success",
+    RequiresRedirect = "requiresRedirect"
+}
+
+enum AuthenticationResultStatus {
+    Redirect = "redirect",
+    Success = "success",
+    Failure = "failure",
+    OperationCompleted = "operation-completed"
+}
+
+interface AuthenticationResult {
+    status: AuthenticationResultStatus;
+    state?: any;
+    message?: string;
+}
+
+interface AuthorizeService {
+    getUser(): Promise<StringDict | undefined>;
+    getAccessToken(request?: AccessTokenRequestOptions): Promise<AccessTokenResult>;
+    signIn(state: any): Promise<AuthenticationResult>;
+    completeSignIn(state: any): Promise<AuthenticationResult>;
+    signOut(state: any): Promise<AuthenticationResult>;
+    completeSignOut(url: string): Promise<AuthenticationResult>;
+}
+
+interface AuthorizeServiceConfiguration extends Msal.Configuration {
+    defaultAccessTokenScopes: string[];
+    additionalScopesToConsent: string[]
+}
+
+class MsalAuthorizeService implements AuthorizeService {
+    _settings: AuthorizeServiceConfiguration;
+    _msalApplication: Msal.UserAgentApplication;
+    _callbackPromise: Promise<AuthenticationResult>;
+
+    constructor(settings: AuthorizeServiceConfiguration) {
```

Comment:
Nit: consider

```suggestion
    constructor(private _settings: AuthorizeServiceConfiguration) {
```

so that L59 is not needed

Reply:
It's not essential but you can also put `readonly` modifiers on the class fields.